src/lib/jsonParseV2/parseEntity.cpp

```diff
@@ -139,34 +140,72 @@ std::string parseEntity(ConnectionInfo* ciP, Entity* eP, bool eidInURL)
       {
         if (type != "String")
         {
-          alarmMgr.badInput(clientIp, "invalid JSON type for entity id");
-          ciP->httpStatusCode = SccBadRequest;;
-          OrionError oe(SccBadRequest, "invalid JSON type for entity id", "BadRequest");
+          const char* errorText = "Invalid JSON type for entity id";
+
+          alarmMgr.badInput(clientIp, errorText);
+          ciP->httpStatusCode = SccBadRequest;
+          OrionError oe(SccBadRequest, errorText, "BadRequest");
+
           return oe.toJson();
         }
 
         eP->id = iter->value.GetString();
+
+        if (forbiddenIdChars(ciP->apiVersion, eP->id.c_str(), ""))
+        {
+          alarmMgr.badInput(clientIp, ERROR_DESC_BAD_REQUEST_INVALID_CHAR_ENTID);
+          ciP->httpStatusCode = SccBadRequest;
+          OrionError oe(SccBadRequest, ERROR_DESC_BAD_REQUEST_INVALID_CHAR_ENTID, "BadRequest");
+
+          return oe.toJson();
+        }
       }
       else  // "id" is present in payload for /v2/entities/<eid> - not a valid payload
       {
-        alarmMgr.badInput(clientIp, "'id' is not a valid attribute");
-        ciP->httpStatusCode = SccBadRequest;;
-        OrionError oe(SccBadRequest, "invalid input, 'id' as attribute", "BadRequest");
+        const char* errorText = "invalid input, 'id' as attribute";
+
+        alarmMgr.badInput(clientIp, errorText);
+        ciP->httpStatusCode = SccBadRequest;
+        OrionError oe(SccBadRequest, errorText, "BadRequest");
+
         return oe.toJson();
       }
     }
     else if (name == "type")
     {
       if (type != "String")
       {
-        alarmMgr.badInput(clientIp, "invalid JSON type for entity type");
+        const char* errorText = "Invalid JSON type for entity type";
+
+        alarmMgr.badInput(clientIp, errorText);
         ciP->httpStatusCode = SccBadRequest;
-        OrionError oe(SccBadRequest, "invalid JSON type for entity type", "BadRequest");
+        OrionError oe(SccBadRequest, errorText, "BadRequest");
+
         return oe.toJson();
       }
 
       eP->type      = iter->value.GetString();
       eP->typeGiven = true;
+
+      if (eP->type.empty())
+      {
+        const char* errorText = "entity type length: 0, min length supported: 1";
```

Comment:
Move to errorMessages.h (in fact, I think the text is already there in a previous define)